detection_rules/devtools.py

```diff
@@ -732,6 +738,61 @@ def update_schemas():
         cls.save_schema()
 
 
+@dev_group.command('update-navigator-gists')
+@click.option('--directory', type=Path, default=CURRENT_RELEASE_PATH.joinpath('extras', 'navigator_layers'),
+              help='Directory containing only navigator files.')
+@click.option('--token', required=True, prompt=get_github_token() is None, default=get_github_token(),
+              help='GitHub token to push to gist', hide_input=True)
+@click.option('--gist-id', default=NAVIGATOR_GIST_ID, help='Gist ID to be updated (must exist).')
+@click.option('--print-markdown', is_flag=True, help='Print the generated urls')
+def update_navigator_gists(directory: Path, token: str, gist_id: str, print_markdown: bool) -> list:
+    """Update the gists with new navigator files."""
```

Comment:
The command itself is written to be generic and update _any_ gist with _any_ directory of navigator json files. We defer to click to handle a lot of that communication, so if a token is missing, it will prompt a user. This includes defaults, meaning if you have a non default, you as a caller should already know the values.

So all of the `Notes` would be overly specific to that one use case.

The defaults are set as they are to simplify calling from the github actions.

Reply:
do you use `gh` CLI tool? We parse that config if it exists as part of the process

https://github.com/elastic/detection-rules/blob/0aeb7399d448186dd01fdb7d7b6d6b776c91e462/detection_rules/devtools.py#L48-L49